examples/04_scalability.py

```diff
@@ -197,3 +200,87 @@ def wrapped_func(*args, **kwargs):
 [t.set(size=16) for t in ax2.get_yticklabels()]
 plt.tight_layout()
 
+##################################################################################
+# SimilarityEncoder with default options and 32 bits float dtype
+# ---------------------------------------------------------------
+#
+# Let us build our vectorizer, using a ColumnTransformer to combine
+# one-hot encoding and similarity encoding.
+# We set the dtype parameters for one-hot and similarity encoders to np.float32
+# and cast the column 'year' to np.float32.
+# Otherwise the ColumnTransformer will force cast everything to float64.
+
+sim_enc = SimilarityEncoder(similarity='ngram', dtype=np.float32, handle_unknown='ignore')
+
+y = df['Violation Type']
+df['Year'] = df['Year'].astype(np.float32)
+
+transformers = [('one_hot', OneHotEncoder(sparse=False, dtype=np.float32, handle_unknown='ignore'),
+                 ['Alcohol',
+                  'Arrest Type',
+                  'Belts',
+                  'Commercial License',
+                  'Commercial Vehicle',
+                  'Fatal',
+                  'Gender',
+                  'HAZMAT',
+                  'Property Damage',
+                  'Race',
+                  'Work Zone']),
+                ('pass', 'passthrough', ['Year']),
+                ]
+
+column_trans = ColumnTransformer(
+    # adding the dirty column
+    transformers=transformers + [('sim_enc', sim_enc, ['Description'])],
+    remainder='drop')
+
+t0 = time()
+X = column_trans.fit_transform(df)
+t1 = time()
+print('Time to vectorize: %s' % (t1 - t0))
+################################################################################
+# We can run a cross-validation
+model = pipeline.make_pipeline(column_trans, log_ref)
+results = resource_used(model_selection.cross_validate)(model, df, y)
+print("Cross-validation score: %s" % results['test_score'])
```

Comment:
Here I would add a small paragraph mentioning that not all scikit-learn models work efficiently with float32, and hence it is worth checking that their indeed is an improvement in resources used.